openid-4-verifiable-credential-issuance-1_0.md

```diff
@@ -710,6 +706,47 @@ Cache-Control: no-store
 }
 ```
 
+# Nonce Endpoint {#nonce-endpoint}
+
+This endpoint allows a Client to acquire a fresh `c_nonce` value without the overhead of a full Credential Request. A Credential Issuer that requires `c_nonce` values to be incorporated into proofs in the Credential Request (see (#credential-request)) MUST offer a Nonce Endpoint.
+
+The `nonce_endpoint` Credential Issuer Metadata parameter, as defined in (#credential-issuer-parameters), contains the URL of the Credential Issuer's Nonce Endpoint.
+
+
+## Nonce Request {#nonce-request}
+
+A request for a nonce is made by sending an HTTP GET request to the URL provided in the `nonce_endpoint` Credential Issuer Metadata parameter.
+
+Below is a non-normative example of a Nonce Request:
+
+```
+GET /nonce HTTP/1.1
```

Comment:
Wouldn't a POST request be more appropriate?

Reply:
+1, I tried to push in this direction as well: https://github.com/openid/OpenID4VCI/pull/381#discussion_r1752363400

Reply:
I didn't realize this was GET. my suggestion has been POST. In german wallet project we have been prototyping with POST and it works well https://bmi.usercontent.opencode.de/eudi-wallet/eidas-2.0-architekturkonzept/flows/PID-IssuerSigned-cloud/#issuer-session-endpoint-at-the-pid-provider

Reply:
> Wouldn't a POST request be more appropriate? 

No.

> I mean the endpoint is supposed to provide a new/different value with every response and the result should not be cached (in my opinion).

That's why it was made not cacheable in the example https://github.com/openid/OpenID4VCI/pull/381/files#diff-1f424614b35a9899813079f1b1f6218631a2aedd993368ccb89bb81a9eda0289R741 and text https://github.com/openid/OpenID4VCI/pull/381/files#diff-1f424614b35a9899813079f1b1f6218631a2aedd993368ccb89bb81a9eda0289R734

Reply:
> +1, I tried to push in this direction as well: [#381 (comment)](https://github.com/openid/OpenID4VCI/pull/381#discussion_r1752363400)

My naive feeling that using a GET to get some data is the most straightforward way to get the data remained unchanged by that discussion over the course of the last two weeks.

Reply:
My 2 cents for what they are worth,

If we asked the question "is this call safe (i.e. is it read-only)" I think the answer would be "maybe". Safe calls shouldn't alter the state of the server in any way but without knowing how the server implements the nonce endpoint we would not know the answer to the question.

If we asked the question "is this call idempotent" I think we would also get a "maybe".  Client could do this call just to "fetch a value" which might not assume any intention of a side-effect, and from the perspective of the server it would depend on the implementation.

I do see the point in making it a GET because it is technically just fetching a random value, but there might be an embedded assumption here that this random value is a prerequisite for some other steps, and that getting a nonce be viewed as setting up a state that needs to be validated by the server.

One could argue that it might be safer to make this a `POST` because this would communicate that the call might not be safe or idempotent and that subsequent calls could cause load, burden, and side-effects for the server.

Reply:
I can see arguments for both as well but was really trying to avoid them entirely with a simple (maybe simplistic) approach.

Reply:
but avoidance didn't work and so f7b4dc8e816a99f8ab3568be14ea62b4446c822f Removed the HTTP method specificity from the Nonce Request text and change the example from GET to POST

Reply:
After reading the comments and listening to the discussions, I'm ok with GET or POST, but we need to pick one.

Reply:
as discussed on the call - it's POST now https://github.com/openid/OpenID4VCI/pull/381/commits/55c8388480cfa5daee7396954283ef34cb606c66